httpgrpc/httpgrpc.go

```diff
@@ -0,0 +1,132 @@
+package httpgrpc
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
+	"github.com/mwitkow/go-grpc-middleware"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sercand/kuberesolver"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/weaveworks/common/middleware"
+)
+
+// HTTPoGRPCServer is a HTTPServer (ie gRPC) implementation
```

Comment:
I don't quite get what the `ie gRPC` implies. I think it means that this is intended to be a gRPC server, but that gRPC servers implement the `HTTPServer` interface. Is that right?